pkg/instance/tshark.go

```diff
@@ -40,13 +42,21 @@ func (i *Instance) createTsharkCollectorInstance(ctx context.Context) (*Instance
 	if err := tsc.Commit(); err != nil {
 		return nil, err
 	}
-	if err := tsc.SetCPU(tsharkCollectorCPU); err != nil {
+	if err := tsc.SetCPU(resource.MustParse(tsharkCollectorCPU)); err != nil {
 		return nil, err
 	}
-	if err := tsc.SetMemory(tsharkCollectorMemory, tsharkCollectorMemory); err != nil {
+	err = tsc.SetMemory(
+		resource.MustParse(tsharkCollectorMemory),
+		resource.MustParse(tsharkCollectorMemory),
+	)
+	if err != nil {
 		return nil, err
 	}
-	if err := tsc.AddVolume(tsharkCollectorVolumePath, i.tsharkCollectorConfig.VolumeSize); err != nil {
+	err = tsc.AddVolume(
+		tsharkCollectorVolumePath,
+		resource.MustParse(i.tsharkCollectorConfig.VolumeSize),
```

Comment:
similar concept to validating at compile, we should push the validation upstream as far as possible. So we should make `tsharkCollectorConfig.VolumeSize` a `resource.Quantity`.

Reply:
oh, nice catch. How could this be swiped out of my hand!